File: Portfolio.Api/Models/PortfolioModels.cs
namespace Portfolio.Api.Models;

public record PersonInfo(
    string Name,
    string Title,
    string Location,
    string Email,
    string Phone,
    string LinkedInUrl,
    string GitHubUrl,
    string Summary
);

public record Skill(string Category, string[] Items);

public record Experience(
    string Company,
    string Role,
    string Period,
    string[] Highlights
);

public record Education(
    string Institution,
    string Degree,
    string Years
);

public record Certification(string Name, string Code);

public record PortfolioData(
    PersonInfo Person,
    Skill[] Skills,
    Experience[] Experiences,
    Education[] Educations,
    Certification[] Certifications
);

public record ContactRequest(string Name, string Email, string Subject, string Message);
public record ContactResponse(bool Success, string Message);

File: portfolio-ui/src/app/models/portfolio.models.ts
export interface PersonInfo {
  name: string;
  title: string;
  location: string;
  email: string;
  phone: string;
  linkedInUrl: string;
  gitHubUrl: string;
  summary: string;
}

export interface Skill {
  category: string;
  items: string[];
}

export interface Experience {
  company: string;
  role: string;
  period: string;
  highlights: string[];
}

export interface Education {
  institution: string;
  degree: string;
  years: string;
}

export interface Certification {
  name: string;
  code: string;
}

export interface PortfolioData {
  person: PersonInfo;
  skills: Skill[];
  experiences: Experience[];
  educations: Education[];
  certifications: Certification[];
}

export interface ContactRequest {
  name: string;
  email: string;
  subject: string;
  message: string;
}

export interface ContactResponse {
  success: boolean;
  message: string;
}
